instrumentation/net/http/otelhttp/transport.go

```diff
@@ -42,9 +42,11 @@ var _ http.RoundTripper = &Transport{}
 
 // NewTransport wraps the provided http.RoundTripper with one that
 // starts a span and injects the span context into the outbound request headers.
-func NewTransport(base http.RoundTripper, opts ...Option) *Transport {
-	t := Transport{
-		rt: base,
+func NewTransport(base http.RoundTripper, opts ...Option) http.RoundTripper {
```

Comment:
Since this is the primary mechanism by which I expect `otelhttp.Transport` instances to be created and this leaves no configuration option for enabling/disabling metric collection, does it make more sense to integrate the metrics collection directly in the `Transport.RoundTrip` method?  This would also be more in line with the approach taken by the `otelhttp.Handler`.

Reply:
I prefer to split up the implementation of metrics and tracing parts for ease of maintenance.